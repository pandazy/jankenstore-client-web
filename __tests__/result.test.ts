import Result from '../src/data/result';
const { ok, err } = Result;

describe('Tests for Result', () => {
	it('Should create ok Result', () => {
		const result = ok(1);
		expect(result.isOk()).toBe(true);
		expect(result.isErr()).toBe(false);
		expect(result.unwrap()).toBe(1);
		expect(() => result.unwrapErr()).toThrowErrorMatchingSnapshot();
	});

	it('Should create err Result', () => {
		const error = new Error('Test error');
		const result = err(error);
		expect(result.isOk()).toBe(false);
		expect(result.isErr()).toBe(true);
		expect(() => result.unwrap()).toThrowErrorMatchingSnapshot();
		expect(result.unwrapErr()).toEqual(error);
	});
});
