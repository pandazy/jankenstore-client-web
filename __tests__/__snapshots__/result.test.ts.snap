// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Result Should create err Result 1`] = `"Test error"`;

exports[`Tests for Result Should create ok Result 1`] = `"unwrapErr called on Ok value: 1"`;
